src/main/webapp/app/exercises/shared/result/updating-result.component.ts

```diff
@@ -172,3 +165,14 @@ export class UpdatingResultComponent implements OnChanges, OnDestroy {
         }
     }
 }
+
+export function getLatestResultOfStudentParticipation(participation: StudentParticipation, showUngradedResults: boolean): Result | undefined {
+    // Sort participation results by completionDate desc.
+    if (participation.results) {
+        participation.results = _orderBy(participation.results, 'completionDate', 'desc');
+    }
+    // The latest result is the first rated result in the sorted array (=newest) or any result if the option is active to show ungraded results.
+    const latestResult = participation.results?.find(({ rated }) => showUngradedResults || rated === true);
+    // Make sure that the participation result is connected to the newest result.
+    return latestResult ? { ...latestResult, participation: participation } : undefined;
+}
```

Comment:
Can you move this into a shard place? This method should be useful for many other places as well.

Reply:
I moved it to the `participation.utils.ts` 👍